helix-core/src/test/java/org/apache/helix/common/ZkTestBase.java

```diff
@@ -123,8 +136,45 @@ public void beforeSuite() throws Exception {
     System.setProperty("zookeeper.4lw.commands.whitelist", "*");
     System.setProperty(SystemPropertyKeys.CONTROLLER_MESSAGE_PURGE_DELAY, "3000");
 
+    // Start an in-memory ZooKeeper
     _zkServer = TestHelper.startZkServer(ZK_ADDR);
     AssertJUnit.assertNotNull(_zkServer);
+    HelixZkClient.ZkClientConfig clientConfig = new HelixZkClient.ZkClientConfig();
+    clientConfig.setZkSerializer(new ZNRecordSerializer());
+    _gZkClient = DedicatedZkClientFactory.getInstance()
+        .buildZkClient(new HelixZkClient.ZkConnectionConfig(ZK_ADDR), clientConfig);
+    _gSetupTool = new ClusterSetup(_gZkClient);
+    _baseAccessor = new ZkBaseDataAccessor<>(_gZkClient);
+    _numZk++; // Now we have 1 ZK
+
+    // If multi-ZooKeeper is enabled, start more ZKs
+    String multiZkConfig = System.getProperty(MULTI_ZK_PROPERTY_KEY);
+    if (multiZkConfig != null && multiZkConfig.equalsIgnoreCase(Boolean.TRUE.toString())) {
+      String numZkFromConfig = System.getProperty(NUM_ZK_PROPERTY_KEY);
+      if (numZkFromConfig != null) {
+        try {
+          int numZkFromConfigInt = Integer.parseInt(numZkFromConfig);
+          // Initialize maps to track multiple ZK servers
+          // Initialize maps to track multiple ZK servers
+          _zkServerMap = new HashMap<>();
+          _helixZkClientMap = new HashMap<>();
+          _clusterSetupMap = new HashMap<>();
+          _baseDataAccessorMap = new HashMap<>();
+          _zkServerMap.put(0, _zkServer);
+          _helixZkClientMap.put(0, _gZkClient);
+          _clusterSetupMap.put(0, _gSetupTool);
+          _baseDataAccessorMap.put(0, _baseAccessor);
```

Comment:
These are duplicate code. Can we just repeatedly call startZooKeeper() and return the tuple that I mentioned above for recording?

Reply:
Removed duplicate code